onadata/apps/api/viewsets/connect_viewset.py

```diff
@@ -114,6 +122,42 @@ def regenerate_auth_token(self, request, *args, **kwargs):
 
         return Response(data=new_token.key, status=status.HTTP_201_CREATED)
 
+    @action(methods=['GET', 'POST'], detail=False)
+    def odk_token(self, request, *args, **kwargs):
+        user = request.user
+
+        if request.method == 'GET':
+            data = {}
+            try:
+                token = ODKToken.objects.get(user=user)
+                expiry_date = token.created + timedelta(
+                    days=ODK_TOKEN_LIFETIME)
+                data.update({'odk_token': token.key})
+                data.update({'expiry_date': expiry_date})
```

Comment:
Could we have it:
```python
data = {
    'odk_token': token.key,
    'expiry_date': expiry_date,
}
```
No need to call any function but have a simple direct assignment.

Reply:
Changed this in the latest commits.